File: traversals/src/test/scala/io/shiftleft/overflowdb/traversals/testdomains/gratefuldead/WrittenBy.scala
package io.shiftleft.overflowdb.traversals.testdomains.gratefuldead

import io.shiftleft.overflowdb._
import scala.jdk.CollectionConverters._

class WrittenBy(graph: OdbGraph, outVertex: NodeRef[ArtistDb], inVertex: NodeRef[SongDb])
  extends OdbEdge(graph, WrittenBy.Label, outVertex, inVertex, WrittenBy.PropertyNames.all.asJava)

object WrittenBy {
  val Label = "writtenBy"

  object Properties {}
  object PropertyNames {
    val all: Set[String] = Set.empty
  }

  val layoutInformation = new EdgeLayoutInformation(Label, PropertyNames.all.asJava)
  var factory: EdgeFactory[WrittenBy] = new EdgeFactory[WrittenBy] {
    override def forLabel(): String = WrittenBy.Label

    override def createEdge(graph: OdbGraph, outNode: NodeRef[OdbNode], inNode: NodeRef[OdbNode]): WrittenBy =
      new WrittenBy(graph, outNode.asInstanceOf[NodeRef[ArtistDb]], inNode.asInstanceOf[NodeRef[SongDb]])
  }
}

File: tinkerpop3/src/test/java/overflowdb/testdomains/gratefuldead/WrittenBy.java
package overflowdb.testdomains.gratefuldead;

import overflowdb.EdgeFactory;
import overflowdb.EdgeLayoutInformation;
import overflowdb.NodeRef;
import overflowdb.OdbGraph;
import overflowdb.Edge;

import java.io.Serializable;
import java.util.Arrays;
import java.util.HashSet;

public class WrittenBy extends Edge implements Serializable {
  public static final String LABEL = "writtenBy";
  public static final HashSet<String> PROPERTY_KEYS = new HashSet<>(Arrays.asList());

  public WrittenBy(OdbGraph graph, NodeRef outVertex, NodeRef inVertex) {
    super(graph, LABEL, outVertex, inVertex, PROPERTY_KEYS);
  }

  public static final EdgeLayoutInformation layoutInformation = new EdgeLayoutInformation(LABEL, PROPERTY_KEYS);

  public static EdgeFactory<WrittenBy> factory = new EdgeFactory<WrittenBy>() {
    @Override
    public String forLabel() {
      return WrittenBy.LABEL;
    }

    @Override
    public WrittenBy createEdge(OdbGraph graph, NodeRef outVertex, NodeRef inVertex) {
      return new WrittenBy(graph, outVertex, inVertex);
    }
  };
}
